### PR TITLE
[auto-materialize] Update `test_asset_daemon.py`

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Changes:
- a lot of repetitive code in this file for making instances with different properties was replaced with a single method for that
- stopped switching between using pytest fixtures and inline functions to create the instances (just use functions, as there isn't much overlap between tests)
- update the AssetDaemonScenario to add more daemon-specific checks when running in "daemon mode". this makes several of the existing tests obsolete, and also helps up our test coverage because we're running these checks against several different scenarios instead of just one.
- run the full end-to-end daemon against an arbitrary 10% of all scenarios. on the fence about this one, I think it's probably fine to just run the end-to-end test against all of the scenarios (just takes a few minutes)

### CODE REVIEW

File: python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/test_asset_daemon.py [hunk 1]
```diff
@@ -1,177 +1,76 @@
...
+        yield instance
 
-    ticks = instance.get_ticks(
-        origin_id=FIXED_AUTO_MATERIALIZATION_ORIGIN_ID,
-        selector_id=FIXED_AUTO_MATERIALIZATION_SELECTOR_ID,
-    )
 
-    # Daemon paused, so no ticks
-    assert len(ticks) == 0
+# take an arbitrary 10% of the scenarios to run end-to-end against the daemon
```
Comment (gibsondan) [inline python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/test_asset_daemon.py:50]: its a little weird that this changes non-deterministically when you add a new scenario in the middle of the list. Did you consider hand-picking a certain number of scenarios instead?

File: python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/test_asset_daemon.py [hunk 2]
```diff
@@ -185,299 +84,114 @@ def _create_tick(instance, status, timestamp):
...
+    with get_daemon_instance(paused=True) as instance:
+        ticks = _get_asset_daemon_ticks(instance)
+        assert len(ticks) == 0
 
-    with pendulum.test(freeze_datetime):
-        scenario = daemon_scenarios["auto_materialize_policy_lazy_freshness_missing"]
-        scenario.do_daemon_scenario(
-            daemon_not_paused_instance,
-            scenario_name="auto_materialize_policy_lazy_freshness_missing",
-        )
```
Comment (gibsondan) [inline python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/test_asset_daemon.py:212]: this was sketchy, sorry

File: python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/test_asset_daemon.py (hunk 1 above)
Comment (sryza) [reply python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/test_asset_daemon.py:50]: yeah I had a similar concern

Comment (OwenKephart) [reply python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/test_asset_daemon.py:50]: hand-picking is definitely reasonable, I think for now I'll just have it test over 100% of the scenarios, and if this step starts taking too long, we can always hand-pick later.

**Comment by OwenKephart** [comment]
@gibsondan actually on second thought, I'll split those updates into a new PR

**Pull request merged**